DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,187 @@
+- [Developer Guide](#developer-guide)
+    - [Getting Started](#getting-started)
+        - [Git Clone logstash-output-opensearch](#git-clone-logstash-output-opensearch-repo)
+        - [Install Prerequisites](#install-prerequisites)
+            - [JRuby](#JRuby)
+            - [Docker](#docker)
+        - [Run Tests](#run-tests)
+        - [Run plugin](#run-plugin-in-logstash)
+        - [Configuration for Logstash Output OpenSearch Plugin](#configuration-for-logstash-output-opensearch-plugin)
+    - [Submitting Changes](#submitting-changes)
+
+# Developer Guide
+
+So you want to contribute code to logstash-output-opensearch? Excellent! We're glad you're here. Here's what you need to do.
+
+## Getting Started
+
+### Fork the logstash-output-opensearch Repo
+
+Fork [opensearch-project/logstash-output-opensearch](https://github.com/opensearch-project/logstash-output-opensearch) and clone locally.
+
+Example:
+```bash
+    git clone https://github.com/[your username]/logstash-output-opensearch.git.
```

Comment:
this should be 'git fork' to match the above text?

Reply:
This is an example for git clone.

Reply:
@VijayanB Ahh - This is why I can't do these at the end of the day. My bad. LGTM